x-pack/plugins/apm/public/components/app/settings/agent_explorer/agent_instances/agent_instances_details/index.tsx

```diff
@@ -0,0 +1,171 @@
+} from '../../../../../shared/managed_table';
+import { TimestampTooltip } from '../../../../../shared/timestamp_tooltip';
+import { TruncateWithTooltip } from '../../../../../shared/truncate_with_tooltip';
+
+type AgentExplorerInstance = ValuesType<
+  APIReturnType<'GET /internal/apm/services/{serviceName}/agent_instances'>['items']
+>;
+
+enum AgentExplorerInstanceFieldName {
+  InstanceName = 'serviceNode',
+  Environments = 'environments',
+  AgentName = 'agentName',
+  AgentVersion = 'agentVersion',
+  LastReport = 'lastReport',
+}
+
+export function getInstanceColumns(
+  serviceName = ''
+): Array<ITableColumn<AgentExplorerInstance>> {
+  return [
+    {
+      field: AgentExplorerInstanceFieldName.InstanceName,
+      name: i18n.translate(
+        'xpack.apm.agentExplorerInstanceTable.InstanceColumnLabel',
+        {
+          defaultMessage: 'Instance',
+        }
+      ),
+      sortable: true,
+      render: (_, { serviceNode }) => {
+        const { displayedName, tooltip } =
+          serviceNode === SERVICE_NODE_NAME_MISSING
+            ? {
+                displayedName: getServiceNodeName(serviceNode),
+                tooltip: i18n.translate(
+                  'xpack.apm.agentExplorerInstanceTable.explainServiceNodeNameMissing',
+                  {
+                    defaultMessage:
+                      'We could not identify the service node. This is likely caused by running a version of APM Server that is older than 7.5. Upgrading to APM Server 7.5 or higher should resolve this issue.',
+                  }
+                ),
+              }
+            : { displayedName: serviceNode, tooltip: serviceNode };
+
+        return (
+          <TruncateWithTooltip
+            data-test-subj="apmAgentExplorerInstanceListServiceLink"
+            text={tooltip}
+            content={
+              <ServiceNodeMetricOverviewLink
+                serviceName={serviceName}
+                serviceNodeName={serviceNode}
+              >
+                <span className="eui-textTruncate">{displayedName}</span>
+              </ServiceNodeMetricOverviewLink>
+            }
+          />
+        );
+      },
+    },
+    {
+      field: AgentExplorerInstanceFieldName.Environments,
+      name: i18n.translate(
+        'xpack.apm.agentExplorerInstanceTable.environmentColumnLabel',
+        {
+          defaultMessage: 'Environment',
+        }
+      ),
+      width: `${unit * 16}px`,
+      sortable: true,
+      render: (_, { environments }) => (
+        <EnvironmentBadge environments={environments ?? []} />
+      ),
+    },
+    {
+      field: AgentExplorerInstanceFieldName.AgentVersion,
+      name: i18n.translate(
+        'xpack.apm.agentExplorerInstanceTable.agentVersionColumnLabel',
+        { defaultMessage: 'Agent Version' }
+      ),
+      width: `${unit * 16}px`,
+      sortable: true,
+      render: (_, { agentVersion }) => (
+        <ItemsBadge
+          items={agentVersion ? [agentVersion] : []}
```

Comment:
Same comment here: I don't think these should be optional? There should always be a version no?